Applications/Spire/Source/KeyBindings/CancelKeyBindingsTableModel.cpp

```diff
@@ -0,0 +1,171 @@
+#include "Spire/KeyBindings/CancelKeyBindingsTableModel.hpp"
+
+using namespace Spire;
+
+namespace {
+  const auto ROW_COUNT = 13;
+  const auto COLUMN_COUNT = 2;
+
+  auto get_action(int index) {
+    static auto actions = std::vector<KeyBindings::CancelAction>({
+      KeyBindings::CancelAction::ALL,
+      KeyBindings::CancelAction::ALL_ASKS,
+      KeyBindings::CancelAction::ALL_BIDS,
+      KeyBindings::CancelAction::CLOSEST_ASK,
+      KeyBindings::CancelAction::CLOSEST_BID,
+      KeyBindings::CancelAction::FURTHEST_BID,
+      KeyBindings::CancelAction::FURTHEST_ASK,
+      KeyBindings::CancelAction::MOST_RECENT,
+      KeyBindings::CancelAction::MOST_RECENT_ASK,
+      KeyBindings::CancelAction::MOST_RECENT_BID,
+      KeyBindings::CancelAction::OLDEST,
+      KeyBindings::CancelAction::OLDEST_ASK,
+      KeyBindings::CancelAction::OLDEST_BID
+    });
+    return actions[index];
+  }
+
+  auto get_index(KeyBindings::CancelAction action) {
+    static auto indexes = std::unordered_map<KeyBindings::CancelAction, int>({
```

Comment:
Similar to `get_action`, using `switch` seems just better. Also, to prevent dangerous duplication of the mapping (one part may be updated while the other update is missing), consider doing something like:
```
case get_action(0):
  return 0;
```
in either function. To avoid runtime performance overhead, make the other function (or both, it does not hurt) `constexpr`.

Reply:
Great idea!